### ISSUE TITLE
routerMeta.load crashes

### ISSUE THREAD

**Issue by SeanCassiere** [TanStack/router#797]
### Describe the bug

When you add a Promise to resolve within the load function of the rootRoute, it throws a react-dom error and crashes the app.

### Your Example Website or App

https://stackblitz.com/edit/tanstack-router-jhbqea?file=src%2Fmain.tsx

### Steps to Reproduce the Bug or Issue

As a promise to be resolved in the `load` fn of the root route.

### Expected behavior

As a user, I expect to be able to resolve a promise in the load fn of the root route similar to the behaviour present in the previous version of the router before the React-only rewrite.

### Platform

- OS: macOS
- Browser: Chrome
- Version: 0.0.1-beta.207

**Comment by SeanCassiere** [comment]
<details><summary>browser error</summary>
<p>
```
Warning: Can't perform a React state update on a component that hasn't mounted yet. This indicates that you have a side-effect in your render function that asynchronously later calls tries to update the component. Move this work to useEffect instead.
    at RouterProvider (http://localhost:3000/node_modules/.vite/deps/chunk-KOYKIKEW.js?v=59c81fce:808:3)
    at RouterWithAuth (http://localhost:3000/src/app-entry.tsx?t=1699308770716:139:27)
    at GlobalDialogProvider (http://localhost:3000/src/hooks/context/modals.tsx:22:45)
    at I18nextProvider (http://localhost:3000/node_modules/.vite/deps/react-i18next.js?v=c47c74d9:800:5)
    at Suspense
    at AuthProvider (http://localhost:3000/node_modules/.vite/deps/react-oidc-context.js?v=afb3a0d8:107:5)
    at QueryClientProvider (http://localhost:3000/node_modules/.vite/deps/chunk-6H2EQ62P.js?v=59c81fce:2587:3)
    at CacheBuster (http://localhost:3000/node_modules/.vite/deps/react-cache-buster.js?v=e3978fa9:132:28)
    at App
``` 
</p>
</details>

**Comment by SeanCassiere** [comment]
With beta.209, this still persists but it doesn't throw an error which is caught and rendered by the Router.

Rather, it permanently suspends are the RouterProvider level.

See the updated Stackblitz repro: https://stackblitz.com/edit/tanstack-router-5zsel5?file=src%2Fmain.tsx

**Comment by manelio** [comment]
Similar problem here. But in my case I'm not loading anything in root route. This is the scenario:

```
function createRouter(app) {
  const rootRoute = rootRouteWithContext()({
    component: () => `Dummy ${app.code}`
  })

  return new Router({ routeTree: rootRoute })
}

const StatefulApp = () => {
  const app = useApp()
  const router = React.useMemo(() => createRouter(app), [app.code])
  return(
    <RouterProvider router={router} />
  )
}
```
useApp is a hook that asynchronously loads app info from the server. Once app state changes, the 

```
 Can't perform a React state update on a component that hasn't mounted yet.
```

Happens.

**Comment by SeanCassiere** [comment]
The original issue seems to have been fixed with `0.0.1-beta.210`
